### PR TITLE
Add withColumnCast to DataFrameExt

### PR DESCRIPTION
**Pull request by nvander1**
Helper to change the datatype of a column

### CODE REVIEW

File: src/main/scala/com/github/mrpowers/spark/daria/sql/DataFrameExt.scala [hunk 1]
```diff
@@ -12,6 +12,32 @@ object DataFrameExt {
...
 
+    /**
+     * Returns a new `DataFrame` with the column `columnName` cast
+     * as `newType`.
+     *
+     * @param columnName the column to cast
+     * @param newType the new type for columnName
+     */
+    def withColumnCast(columnName: String, newType: String): DataFrame =
+      df.select((df.columns.map {
```
Comment (manuzhang) [inline src/main/scala/com/github/mrpowers/spark/daria/sql/DataFrameExt.scala]: It seems this map function can be abstracted out as a more general way to update one column

**Comment by MrPowers** [comment]
@nvander1 - this looks great.  Can we change this method name to something like `castColumn`?  I typically only use `with` at the beginning of method names when a column is being added.  See [this part of the spark-style-guide](https://github.com/MrPowers/spark-style-guide#naming-conventions).  Let me know what you think!

**Comment by nvander1** [comment]
I think the name withColumnCast is consistent with the built in
withColumnRenamed. But both sound clear to me.

Le mer. 22 mai 2019 à 14:46, Matthew Powers <notifications@github.com> a
écrit :

**Comment by MrPowers** [comment]
@nvander1 - That's a good point.  Going to merge this in.  This function will be really useful.

**Pull request merged**